keyvi/src/cpp/dictionary/dictionary.h

```diff
@@ -40,11 +40,31 @@ namespace dictionary {
 class Dictionary
 final {
    public:
-    Dictionary(const char* filename, bool load_lazy = false)
+
+  /**
+   * DEPRECATED: Initialize a dictionary in lazy or non-lazy mode.
+   *
+   * Use Dictionary(filename, loading_strategy) instead.
+   *
+   * @param filename the filename
+   * @param load_lazy whether to load lazy.
+   */
+    Dictionary(const char* filename, bool load_lazy)
         : fsa_(new fsa::Automata(filename, load_lazy)) {
       TRACE("Dictionary from file %s", filename);
     }
 
+    /**
+     * Initialize a dictionary from a file.
+     *
+     * @param filename filename to load keyvi file from.
+     * @param loading_strategy optional: Loading strategy to use.
+     */
+    Dictionary(const char* filename, loading_strategy_types loading_strategy = loading_strategy_types::lazy)
+       : fsa_(new fsa::Automata(filename, loading_strategy)) {
```

Comment:
Instead of `new` operator, we can use `std::make_shared`, it is doing just one allocation for counter and actual object, and has only 1 pointer instead of 2.